python/cuml/test/test_tsne.py

```diff
@@ -44,7 +44,77 @@ def check_embedding(X, Y, score=0.76):
 
 
 @pytest.mark.parametrize('name', dataset_names)
-def test_tsne(name):
+@pytest.mark.parametrize('type_knn_graph', ['sklearn', 'cuml'])
+@pytest.mark.parametrize('method', ['fft', 'barnes_hut'])
+def test_tsne_knn_graph_used(name, type_knn_graph, method):
+
+    datasets
+    X = eval("datasets.load_{}".format(name))().data
```

Comment:
Ok, I think I see whats going on. Using `eval` is very awkward and not recommended. It would be much better to parameterize the load functions instead of building up the function name. Something like this should work:

```python
dataset_load_functions = [
    datasets.load_digits,
    datasets.load_boston,
    datasets.load_iris,
    datasets.load_breast_cancer,
    datasets.load_diabetes,
]
```

Then you could easily call it with:
```python
@pytest.mark.parametrize('load_dataset', dataset_load_functions)
@pytest.mark.parametrize('type_knn_graph', ['sklearn', 'cuml'])
@pytest.mark.parametrize('method', ['fft', 'barnes_hut'])
def test_tsne_knn_graph_used(load_dataset, type_knn_graph, method):

    X = load_dataset().data
```